nxdrive/gui/api.py

```diff
@@ -315,7 +344,19 @@ def open_direct_transfer(self, uid: str) -> None:
         self.application.hide_systray()
         engine = self._manager.engines.get(uid)
         if engine:
-            self.application.show_server_folders(engine, None)
+            direct_transfers = [
+                dt
+                for dt in list(engine.dao.get_uploads())
+                if self.upload_is_direct_transfer(dt.path, engine)
+            ]
+            if direct_transfers:
+                self.application.show_direct_transfer_window(engine.uid)
+            else:
+                self.application.show_server_folders(engine, None)
+
+    @pyqtSlot()
+    def close_direct_transfer(self) -> None:
```

Comment:
Missing doctstring + types + changelog.
```suggestion
    def close_direct_transfer_windows(self) -> None:
```